crates/astria-composer/src/config/searcher.rs

```diff
@@ -0,0 +1,119 @@
+use std::{
+    net::{
+        AddrParseError,
+        SocketAddr,
+    },
+    str::FromStr,
+};
+
+use figment::{
+    providers::{
+        Env,
+        Serialized,
+    },
+    Figment,
+};
+use serde::{
+    Deserialize,
+    Serialize,
+};
+use thiserror::Error;
+
+use super::cli;
+use crate::types::rollup::ChainId;
+
+const DEFAULT_SEQUENCER_URL: &str = "127.0.0.1:1317";
+const DEFAULT_API_URL: &str = "127.0.0.1:8080";
+const DEFAULT_CHAIN_ID: &str = "ethereum";
+const DEFAULT_EXECUTION_RPC_URL: &str = "127.0.0.1:50051";
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("invalid config")]
+    ConfigExtraction(#[from] figment::Error),
+    #[error("invalid api_url")]
+    InvalidApiUrl(#[from] AddrParseError),
+    #[error("missing cli field")]
+    MissingCliField(String),
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq, Clone)]
+pub struct Config {
+    /// Address of the API server
+    #[serde(default = "default_api_url")]
+    pub api_url: SocketAddr,
+
+    /// Address of the RPC server for the sequencer chain
+    #[serde(default = "default_sequencer_url")]
+    pub sequencer_url: SocketAddr,
+
+    /// Chain ID that we want to connect to
+    #[serde(default = "default_chain_id")]
+    pub chain_id: ChainId,
+
+    /// Address of the RPC server for execution
+    #[serde(default = "default_execution_rpc_url")]
+    pub execution_rpc_url: SocketAddr,
```

Comment:
What are we executing? 

I'm guessing the idea here is like as a block builder, but this might require some rethinking of APIs implemented on sequenced chains. The execution API drives the chain and adds blocks to the chain, not the idea here presumably. That would be work done by conductor.

Reply:
will be using this to initialize an ethers::Providerwebsocket connection (the actual value will probably change, this is just a placeholder copied over from conductor)

Reply:
https://github.com/astriaorg/astria/pull/125#discussion_r1248066300

Reply:
As noted above I think we should rename this, since this is a eth_websocket connection

Reply:
renamed to `execution_ws_url` and changed the port to 8546